sdk/cosmos/azure-cosmos/src/main/java/com/azure/cosmos/implementation/directconnectivity/GoneAndRetryWithRetryPolicy.java

```diff
@@ -231,7 +231,7 @@ public Mono<ShouldRetryResult> shouldRetry(Exception exception) {
                     exception);
 
                 return Mono.just(ShouldRetryResult.noRetry(exceptionToThrow,
-                    Quadruple.with(true, false, Duration.ofMillis(0), this.attemptCount.get())));
+                    Quadruple.with(true, true, Duration.ofMillis(0), this.attemptCount.get())));
```

Comment:
QQ - what is the second value used for, somehow did not really find usage of it

Reply:
The second boolean maps to the `inRetry` argument here - https://github.com/Azure/azure-sdk-for-java/blob/main/sdk/cosmos/azure-cosmos/src/main/java/com/azure/cosmos/implementation/directconnectivity/ConsistencyReader.java#L176-L200. It is to gate against a particular timer when `ConsistencyReader#readAsync` is being in-region retried in its entirety. For 410-1022, I'd argue the second boolean doesn't matter as `ConsistencyReader#readAsync` won't be in-region retried.